punishments.js

```diff
@@ -1268,10 +1327,10 @@ Punishments.monitorRoomPunishments = function (user) {
 		if (Config.punishmentautolock && points >= AUTOLOCK_POINT_THRESHOLD) {
 			let rooms = punishments.map(([room]) => room).join(', ');
 			let reason = `Autolocked for having punishments in ${punishments.length} rooms: ${rooms}`;
-			let message = `${user.name} was locked for having punishments in ${punishments.length} rooms: ${punishmentText}`;
+			let message = `${user.name || toId(user)} was locked for having punishments in ${punishments.length} rooms: ${punishmentText}`;
```

Comment:
It is if user is a string and not an object, which is the case when the user gets autolocked for a punishment issued while the user is offline.